dkroutingtool/src/py/optimization.py

```diff
@@ -1454,7 +1508,46 @@ def segment_unload_route(route):
                     del routes_for_mapping[zone_route]
                     del vehicles[zone_route]
                     solution.zone_route_map[zone_to_segment] = zone_routes
-                    
+
+    if color_naming:
+        new_routes_for_mapping = {}
+        new_vehicles = {}
+        for key in routes_for_mapping:
+            if '-' in key:
+                number_key = key.split('-')[0]
+                second_part = key.split('-')[1]
+            else:
+                number_key = key
+                second_part = ''
+            
+            new_key = display_dict[number_key]
+            if second_part != '':
+                new_key = f'{new_key}-{second_part}'
+
+            new_routes_for_mapping[new_key] = routes_for_mapping[key]
+            new_vehicles[new_key] = vehicles[key]
+        
+        routes_for_mapping = new_routes_for_mapping
+        vehicles = new_vehicles
+
+        for key in solution.zone_route_map:
+            new_routes = []
+            for route in solution.zone_route_map[key]:
+                if '-' in route:
```

Comment:
this logic is repeated as above. I do find it nervous to encode the keys in this way. It might be a bit cleaner to separate it into two functions used here and in the original:

key = encode_key(number, name)
number, name = decode_key(key_str)

Reply:
The abuse of keys is real, and I quickly tried to fix it to have a displayable name and a key but got exhausted jumping through hoops so everything would be clean so I ended up with this solution. The abuse of dictionaries is also a thing, haha, so I'll open a ticket that asks for a rejig of how we keep track of solution, creating a well-defined struct for it.